dev-tools/omdb/src/bin/omdb/db.rs

```diff
@@ -504,6 +522,153 @@ async fn cmd_db_disk_info(
+    limit: NonZeroU32,
+    args: &DiskPhysicalArgs,
+) -> Result<(), anyhow::Error> {
+    // We start by finding any zpools that are using the physical disk.
+    use db::schema::zpool::dsl as zpool_dsl;
+    let zpools = zpool_dsl::zpool
+        .filter(zpool_dsl::time_deleted.is_null())
+        .filter(zpool_dsl::physical_disk_id.eq(args.uuid))
+        .select(Zpool::as_select())
+        .load_async(datastore.pool_for_tests().await?)
+        .await
+        .context("loading zpool from pysical disk id")?;
+
+    let mut sled_ids = HashSet::new();
+    let mut dataset_ids = HashSet::new();
+
+    // The current plan is a single zpool per physical disk, so we expect that
+    // this will have a single item.  However, If single zpool per disk ever
+    // changes, this code will still work.
+    for zp in zpools {
+        // zpool has the sled id, record that so we can find the serial number.
+        sled_ids.insert(zp.sled_id);
+
+        // Next, we find all the datasets that are on our zpool.
+        use db::schema::dataset::dsl as dataset_dsl;
+        let datasets = dataset_dsl::dataset
+            .filter(dataset_dsl::time_deleted.is_null())
+            .filter(dataset_dsl::pool_id.eq(zp.id()))
+            .select(Dataset::as_select())
+            .load_async(datastore.pool_for_tests().await?)
+            .await
+            .context("loading dataset")?;
+
+        // Add all the datasets ids that are using this pool.
+        for ds in datasets {
+            dataset_ids.insert(ds.id());
+        }
+    }
+
+    // If we do have more than one sled ID, then something is wrong, but
+    // go ahead and print out whatever we have found.
+    for sid in sled_ids {
+        let (_, my_sled) = LookupPath::new(opctx, datastore)
+            .sled_id(sid)
+            .fetch()
+            .await
+            .context("failed to look up sled")?;
+
+        println!(
+            "Physical disk: {} found on sled: {}",
+            args.uuid,
+            my_sled.serial_number()
+        );
+    }
+
+    let mut volume_ids = HashSet::new();
+    // Now, take the list of datasets we found and search all the regions
+    // to see if any of them are on the dataset.  If we find a region that
+    // is on one of our datasets, then record the volume ID of that region.
+    for did in dataset_ids.clone().into_iter() {
+        use db::schema::region::dsl as region_dsl;
+        let regions = region_dsl::region
+            .filter(region_dsl::dataset_id.eq(did))
+            .select(Region::as_select())
+            .load_async(datastore.pool_for_tests().await?)
+            .await
+            .context("loading region")?;
+
+        for rs in regions {
+            volume_ids.insert(rs.volume_id());
+        }
+    }
+
+    // At this point, we have a list of volume IDs that contain a region
+    // that is part of a dataset on a pool on our disk.  The final step is
+    // to find the virtual disks associated with these volume IDs and
```

Comment:
Unrelated to this change, is there a diagram somewhere showing how disks, volumes, and regions are related?

Reply:
I don't think a diagram exists.  Part of why I did this was to figure out what that 
mapping was.  If I make a diagram, where would we put it?

Reply:
I'm not sure.  Could be a new doc in "./docs" here, or in the Crucible repo, or in some Crucible-related RFD?

Reply:
How about this: https://rfd.shared.oxide.computer/rfd/0177#_relationship_between_volumes_and_api_objects

Reply:
That's very helpful!  Thanks.